manifests/server/role.pp

```diff
@@ -4,7 +4,6 @@
   $password_hash    = false,
   $createdb         = false,
   $createrole       = false,
-  $db               = $postgresql::server::default_database,
```

Comment:
This parameter is considered public API and removing it would be breaking, even if it doesn't actually do anything. If the db param is not needed then we could just make it unused and deprecated it, and remove it later.

Does the database specified not affect the `ALTER ROLE` commands below? Are roles always defined globally and not db-specific?

Reply:
Interesting that it was not documented. Yes, I think it should still be kept and just emit a warning and do nothing. It has been there ever since `postgresql::server::role` was created in 2013 and even before that for `postgresql::role`